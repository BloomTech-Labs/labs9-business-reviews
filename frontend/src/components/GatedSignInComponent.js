import React, { Component } from 'react';
import axios from 'axios';
import LoginForm from './LoginForm';
import { backendLink } from '../assets/config';

export default class GatedSignIn extends Component {
  constructor(props) {
    super(props);
    this.state = { user: {} };
  }

  async componentDidMount() {
    const res = await axios.get(`${backendLink}/api/user/me`, {
      withCredentials: 'include'
    });
    if (!res.data.user) return null;
    const [user] = res.data.user;
    this.setState({ user });
  }
  render() {
    return (
      <>{this.state.user.name ? <>{this.props.children}</> : <LoginForm {...this.props} />}</>
    );
  }
}
