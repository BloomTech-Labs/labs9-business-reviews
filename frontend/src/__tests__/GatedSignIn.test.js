import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import GatedSignInComponent from '../components/GatedSignInComponent';

describe('<GatedSignInComponent />', () => {
  it('renders', () => {
    shallow(<GatedSignInComponent />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<GatedSignInComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
