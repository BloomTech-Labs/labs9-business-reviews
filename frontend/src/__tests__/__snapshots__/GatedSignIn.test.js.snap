// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GatedSignInComponent /> matches snapshot 1`] = `
<Fragment>
  <LoginForm />
</Fragment>
`;
